cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/server.go

```diff
@@ -49,14 +52,18 @@ type Server struct {
 	// clientGetter is a field so that it can be switched in tests for
 	// a fake client. NewServer() below sets this automatically with the
 	// non-test implementation.
-	clientGetter func(context.Context) (dynamic.Interface, error)
+	clientGetter    func(context.Context, kube.ClustersConfig, bool) (dynamic.Interface, error)
+	clustersConfig  kube.ClustersConfig
+	unsafeUseDemoSA bool
 }
 
 // NewServer returns a Server automatically configured with a function to obtain
 // the k8s client config.
-func NewServer() *Server {
+func NewServer(config kube.ClustersConfig, useSA bool) *Server {
 	return &Server{
-		clientGetter: clientForRequestContext,
+		clientGetter:    clientForRequestContext,
+		clustersConfig:  config,
+		unsafeUseDemoSA: useSA,
```

Comment:
I don't like that part, I need the `useSA` and `config` in the server to pass the information via `v1alpha1.RegisterKappControllerPackagesServiceServer(s, NewServer(config, unsafeUseDemoSA))` in the  `RegisterWithGRPCServer` function. Currently, it's the only way I came up with, but I feel I'm missing something

Reply:
Right, and more importantly, then each plugin is responsible for creating the client. Have a play with my suggestion above, if you get a chance.